provider/data_source_keycloak_user.go

```diff
@@ -0,0 +1,38 @@
+package provider
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func dataSourceKeycloakUser() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceKeycloakUserRead,
+		Schema: map[string]*schema.Schema{
+			"realm_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func dataSourceKeycloakUserRead(data *schema.ResourceData, meta interface{}) error {
+	keycloakClient := meta.(*keycloak.KeycloakClient)
+
+	realmId := data.Get("realm_id").(string)
+	username := data.Get("username").(string)
+
+	user, err := keycloakClient.GetUserByUsername(realmId, username)
+	if err != nil {
+		return err
+	}
+
+	mapFromUserToData(data, user)
```

Comment:
Once master is merged into this branch, this line will cause test failures, since this function will attempt to set attributes that aren't defined in the data source.

Reply:
Changed as requested